feat(Browser): Clarified browser install doc to include four installation methods (#18074)
newrelic/docs-website: src/content/docs/browser/browser-monitoring/installation/install-browser-monitoring-agent.mdx

## Before you begin [#before-start] 

If you haven't done so, you may want to check that your browser app meets our [compatibility and requirements](/docs/browser/new-relic-browser/getting-started/compatibility-requirements-browser-monitoring).

## Install the browser agent [#install]

There are four ways to install the browser agent. Depending on your specific needs, you can:

* Add the snippet directly to your site's code
* Use one of our APM agents to inject the code for you
* Query NerdGraph, our GraphQL API, to configure and retrieve the snippet
* Instrument multiple apps at once with npm and NerdGraph

<Tabs>
	<TabsBar>
        <TabsBarItem id="1">
            Add snippet
        </TabsBarItem>
        <TabsBarItem id="2">
            Install via APM
        </TabsBarItem>
        <TabsBarItem id="3">
            Use NerdGraph
        </TabsBarItem>
        <TabsBarItem id="4">
            Use npm
        </TabsBarItem>
    </TabsBar>

    <TabsPages>
        <TabsPageItem id="1">
## Add the snippet directly [#add-snippet]

<img
  title="Install browser"
  alt="Install browser"
  src={browserInstallBrowser}
/>

> Go to <DNT>**[one.newrelic.com](https://one.newrelic.com) > Add data > Browser and mobile > Browser monitoring**</DNT> and follow the instructions to generate the browser JavaScript snippet in the UI and add it to your site's code. 

Reviewer: ```suggestion
Go to <DNT>**[one.newrelic.com](https://one.newrelic.com) > Integrations & Agents > Browser and mobile > Browser monitoring**</DNT> and follow the instructions to generate the browser JavaScript snippet in the UI and add it to your site's code. 
```

Reviewer: This is a new change introduced in https://github.com/newrelic/docs-website/pull/18021

Author: Oh, of course! Thanks for catching this @ally-sassman!

Author: Resolved in my latest commit

<ButtonGroup>
  <ButtonLink
    role="button"
    to="https://onenr.io/0MRNpYqNqQn"
    variant="primary"
  >
    Get your browser snippet
  </ButtonLink>
</ButtonGroup>

Once you have the snippet, paste it as close to the top of your `<head>` tag as possible, but after any position-sensitive META tags (such as `X-UA-Compatible` and `charset`). 

For more detailed information about the placing the snippet, see [JavaScript placement requirements](/docs/browser/new-relic-browser/page-load-timing-resources/instrumentation-browser-monitoring#javascript-placement).

<Collapser
    id="browser_delay"
    title="Delay browser monitoring"
>

<Callout variant="caution">

>   Delaying instantiation causes the agent to miss information until it is fully loaded and initialized. The order of execution of these scripts does lead to some loss in instrumentation data from the start of page views.

Reviewer: ```suggestion
  Delaying instantiation causes the agent to miss information until it's fully loaded and initialized. The order of execution of these scripts does lead to some loss in instrumentation data from the start of page views.
```

Author: Resolved in my latest commit

Once the agent is fully initialized, it functions as expected. Any delay also affects when the agent wraps globals or subscribes to events relative to other libraries or code that affects captured data.

You can delay the agent initialization in these ways:

* Add a `script-src` tag with the `async` attribute.
* Add a `script-src` tag with the `defer` attribute.
* Use dynamic import or chronologically late placement of the loader code.

Your page's `src` must point to a valid agent loader version in our CDN. For example, the URL that always contains our current Pro+SPA agent release is `https://js-agent.newrelic.com/nr-loader-spa-current.min.js`. [CSP is required](/docs/browser/new-relic-browser/getting-started/compatibility-requirements-browser-monitoring#csp) to load external third-party scripts.

To determine when you can start calling the `newrelic` API, subscribe to an event at the global scope:

```js
window.addEventListener('newrelic', (evt) => {
  if (evt.detail.loaded) {
    // you can start using newrelic.interaction(), etc. now
  }
})
```
    
</Collapser>
        </TabsPageItem>
        <TabsPageItem id="2">
## Inject browser agent via APM [#install-apm]

> With an APM agent, you can deploy browser monitoring without manually adding the snippet to your code.

Reviewer: I would remove this sentence since it's basically repeated in the following sentences

Author: I ended up rewriting these two paragraphs to address this

* [PHP](/docs/agents/php-agent/features/page-load-timing-php)
* [Python](/docs/agents/python-agent/supported-features/page-load-timing-python)
* [Ruby](/docs/agents/ruby-agent/features/page-load-timing-ruby)
        </TabsPageItem>
        <TabsPageItem id="3">
## Query NerdGraph for the browser snippet [#install-nerdgraph]

> If you'd prefer to create your browser snippet programmatically, you can do so via our GraphQL API, NerdGraph. 

Reviewer: ```suggestion
If you'd prefer to create your browser snippet programmatically, you can do so via NerdGraph, our GraphQL API.
```

Author: Resolved in my latest commit

## Instrument multiple apps via npm [#install-npm]

If you want to automate browser monitoring deployment across multiple web apps, you can use NerdGraph and our browser agent npm package to configure and deploy the agent as part of your application build process.

For more on this, see [instrument multiple apps with npm and NerdGraph](/docs/apis/nerdgraph/examples/combining-npm-nerdgraph/).
        </TabsPageItem>
    </TabsPages>
</Tabs>

## What's next? [#whats-next]

When you've installed the browser agent, wait a few minutes and look for the data in [our browser monitoring UI](https://one.newrelic.com/nr1-core?filters=%28domain%3D%27BROWSER%27%20AND%20type%3D%27APPLICATION%27%29&state=1f7bae2d-dd3a-0027-3079-b60bcd19ded9&_gl=1*15f78ub*_ga*MTE0MTA2MTA5MS4xNzE4OTI0MDkw*_ga_R5EF3MCG7B*MTcyMTMxMzk2MC41Ny4xLjE3MjEzMTg2MzcuNDcuMS4xMDYwNTI3Njcw).

If it's been more than 15 minutes and you're not seeing data, see [troubleshoot your browser installation](/docs/browser/browser-monitoring/troubleshooting/troubleshoot-your-browser-monitoring-installation/).

Once you're seeing browser data in the UI, here are some next steps on how you can get the most out that data: 

> * View your app data on the [summary page](/docs/browser/browser-monitoring/getting-started/browser-summary-page).

Reviewer: ```suggestion
* [View a high-level summary of your app data](/docs/browser/browser-monitoring/getting-started/browser-summary-page)
```

> * Now that you've instrumented with the browser agent, go deeper with your data with our [improve your website tutorial](/docs/journey-performance/guide-to-monitoring-core-web-vitals). 

Reviewer: ```suggestion
* [Monitor your core web vitals](/docs/tutorial-improve-site-performance/guide-to-monitoring-core-web-vitals/)
```

Reviewer: Either use this updated core web vitals link, or link to the first page in that tutorial: https://docs.newrelic.com/docs/tutorial-improve-site-performance/improve-website-performance/

> * Learn how to optimize your browser monitoring experience with [best practices guide for browser monitoring](/docs/new-relic-solutions/best-practices-guides/full-stack-observability/browser-monitoring-best-practices-guide). 

Reviewer: ```suggestion
* [Learn how to optimize your browser monitoring experience](/docs/new-relic-solutions/best-practices-guides/full-stack-observability/browser-monitoring-best-practices-guide)
```
